src/components/quickReplyModal/quickReplyModalContent.tsx

```diff
@@ -8,43 +8,85 @@ import { useSelector, useDispatch } from 'react-redux';
 import { delay, generateReplyPermlink } from '../../utils/editor';
 import { postComment } from '../../providers/hive/dhive';
 import { toastNotification } from '../../redux/actions/uiAction';
-import { updateCommentCache } from '../../redux/actions/cacheActions';
+import {
+  deleteQuickCommentCacheEntry,
+  updateCommentCache,
+  updateQuickCommentCache,
+} from '../../redux/actions/cacheActions';
 import { default as ROUTES } from '../../constants/routeNames';
 import get from 'lodash/get';
 import { navigate } from '../../navigation/service';
 import { postBodySummary } from '@ecency/render-helper';
+import { QuickComment } from '../../redux/reducers/cacheReducer';
 
 export interface QuickReplyModalContentProps {
   fetchPost?: any;
   selectedPost?: any;
   inputRef?: any;
   sheetModalRef?: any;
+  handleCloseRef?: any;
 }
 
 export const QuickReplyModalContent = ({
   fetchPost,
   selectedPost,
   inputRef,
   sheetModalRef,
+  handleCloseRef,
 }: QuickReplyModalContentProps) => {
   const intl = useIntl();
   const dispatch = useDispatch();
   const currentAccount = useSelector((state) => state.account.currentAccount);
   const pinCode = useSelector((state) => state.application.pin);
+  const quickComments = useSelector((state) => state.cache.quickComments);
 
   const [commentValue, setCommentValue] = useState('');
   const [isSending, setIsSending] = useState(false);
+  const [quickCommentCache, setQuickCommentCache] = useState<QuickComment>(null);
 
   const headerText =
     selectedPost && (selectedPost.summary || postBodySummary(selectedPost, 150, Platform.OS));
+  const parentAuthor = selectedPost ? selectedPost.author : '';
+  const parentPermlink = selectedPost ? selectedPost.permlink : '';
+  const path = `${parentAuthor}/${parentPermlink}`;
 
-  // reset the state when post changes
   useEffect(() => {
-    setCommentValue('');
+    handleCloseRef.current = handleSheetClose;
+  }, [commentValue]);
+
+  // load quick comment value from cache
+  useEffect(() => {
+    if (quickComments.has(path)) {
+      const quickComment: QuickComment = quickComments.get(path);
+      setCommentValue(quickComment.body);
+      setQuickCommentCache(quickComment);
+    } else {
+      setCommentValue('');
+    }
   }, [selectedPost]);
 
   // handlers
 
+  const handleSheetClose = () => {
+    _addQuickCommentIntoCache();
+  };
+
+  // add quick comment value into cache
+  const _addQuickCommentIntoCache = () => {
+    const date = new Date();
+    const updatedStamp = date.toISOString().substring(0, 19);
+
+    const quickCommentCacheData = {
+      parent_permlink: parentPermlink,
```

Comment:
instead of using `parent_permlink`, use simple prop `id` with value `${parentAuthor}/${parentPermlink}` .... this will simplify it's scalability with post drafts cache... check `setDraftPost` method in `realm.js`

Reply:
@aliseyalvi `parent_permlink` is not serving any purpose in current arrangement, perhaps can be removed to avoid confusion

Reply:
updated